### PR TITLE
Problem-Set-2_Script

### PR DESCRIPTION
**Pull request by casey-martin**

### CODE REVIEW

**Comment by casey-martin** [comment]
Thanks for the notice! answers.yml should be up to date.

File: answers.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	| grep -v "^[0-9]_")
+
+
+echo answer-2: $myanswer2
+#rm /tmp/answer2.bed
+
+
+# is it more efficient to filter out non-CTCF peaks prior to bedtools 
+# operations? if so, is speed up worth it? need to find out why 
+# bedtools indexing is better than nested "for loops"
```
Comment (kriemo) [inline answers.sh]: FYI, if you are interested in the algorithms, bedtools uses a binning algorithm described [here](http://genome.cshlp.org/content/12/6/996.long) by default or a chromosome sweep algorithm described [here](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0973-5) depending on whether the `-sorted` flag is used.

File: answers.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+echo answer-1: $myanswer1
+
+#generate temporary bed file with interval specified. current script leaves bed file in tmp
+#for inspection in case of debugging. uncomment the "rm" command to delete tmp file.
+
+echo -e "chr22\t19000000\t19000500\tfoo\n" > /tmp/answer2.bed
+
+myanswer2=$(bedtools nuc -fi $mydir/fasta/hg19.chr22.fa -bed /tmp/answer2.bed \
+	| cut -f 6 \
+	| grep -v "^[0-9]_")
```
Comment (kriemo) [inline answers.sh]: nice use of a regular expression.

File: answers.sh (hunk 1 above)
Comment (casey-martin) [inline answers.sh]: For the binning, is it essentially fitting the search space to match the size of the query?

Comment (kriemo) [inline answers.sh]: Essentially yes. See this biostars post if you are interested in more details about these types of algorithms (https://www.biostars.org/p/2244/)

**Pull request closed without merging**